compiler/src/main/kotlin/interpreter/interpreter.kt

```diff
@@ -0,0 +1,690 @@
+package dk.aau.cs.d409f19.cellumata.interpreter
+
+import dk.aau.cs.d409f19.cellumata.ast.*
+import dk.aau.cs.d409f19.cellumata.visitors.ASTVisitor
+import java.awt.Color
+import java.awt.Dimension
+import java.awt.Graphics2D
+import java.lang.RuntimeException
+import java.util.*
+import javax.swing.JFrame
+import javax.swing.JPanel
+
+open class InterpretRuntimeException(msg: String) : RuntimeException(msg)
+class InterpreterVisitException(node: Any) : InterpretRuntimeException("${node.javaClass} should not be visited.")
+
+/**
+ * Custom values used by the interpreter. StateValue is a state and a index (used for multi-states).
+ * When a state is mentioned in the source code, e.g. Alive, a StateValue equal to (Alive, 0) is created.
+ * The LookupExpr affects the index of the state value, e.g. Alive[2] = (Alive, 2)
```

Comment:
```suggestion
 * The LookupExpr affects the index of the state value, e.g. `Alive[2] = (Alive, 2)`
```
[Square brackets reference other elements in Kotlin docstrings](https://kotlinlang.org/docs/reference/kotlin-doc.html). Use inline-code markdown for proper viewing, as escaping the brackets don't work.

Reply:
Added in ca0e7fc